src/Controls/src/Core/Setter.cs

```diff
@@ -104,8 +104,9 @@ internal void UnApply(BindableObject target, bool fromStyle = false)
 				return;
 
 			object actual = targetObject.GetValue(Property);
-			if (!Equals(actual, Value) && !(Value is Binding) && !(Value is DynamicResource))
+			if (!Equals(actual, Value) && Value is not BindingBase && Value is not DynamicResource)
```

Comment:
that's it. that's the fix. replacing Binding by BindingBase. 4 chars. I replaced some statement with pattern matching to obfuscate the stupidest fix ever and so I don't look as dumb as I really am.